tests/queryset_pickle/tests.py

```diff
@@ -212,6 +212,12 @@ def test_annotation_with_callable_default(self):
         qs = Happening.objects.annotate(latest_time=models.Max('when'))
         self.assert_pickles(qs)
 
+    def test_filter_deferred(self):
+        qs = Happening.objects.all()
+        qs._defer_next_filter = True
+        qs = qs.filter(id=0)
```

Comment:
This test passes even without the patch applied. Did you mean to use the id of the `Happening` created in `setUpTestData`? I guess you could assign the `cls.happening` there and use `self.happening.id` here.

Reply:
> Did you mean to use the id of the Happening created in setUpTestData?

No, I was actually worried that list(Happing.object.all()) would be equivalent to list(Happening.objects.filter(id=...)), so test would pass even in pickle lost the deferred filter.
So I intentionally created empty queryset.

Reply:
Ahh I see, the test is the other way around; if the filter wasn't applied all results would be returned 👍